src/core/server/server.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { first } from 'rxjs/operators';
+import { ConfigService, Env } from './config';
+import { ElasticsearchModule } from './elasticsearch';
+import { HttpConfig, HttpModule, HttpServerInfo } from './http';
+import { LegacyCompatModule } from './legacy';
+import { Logger, LoggerFactory } from './logging';
+import { PluginsModule } from './plugins';
+
+export class Server {
```

Comment:
Can we go ahead and make this analogous class have the same name as the frontend: `CoreSystem`?

Reply:
I think `src/core/server/root` is more like the "CoreSystem" equivalent in /server. Perhaps it's better to wait till we combine these classes and then call the result `CoreSystem`?